File: HW 6 (Multithreading)/customview/src/main/java/com/example/customview/MainActivity.java
package com.example.customview;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.google.android.material.snackbar.BaseTransientBottomBar;
import com.google.android.material.snackbar.Snackbar;

import static com.example.customview.NOTIFICATION_TYPE.SNACKBAR;
import static com.example.customview.NOTIFICATION_TYPE.TOAST;

public class MainActivity extends AppCompatActivity {

    private Toolbar toolbar;

    private NotificationManager notificationManager = NotificationManager.getNotificationManager(this);
    private NOTIFICATION_TYPE notifcationType;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        notifcationType = NOTIFICATION_TYPE.valueOf(notificationManager.loadChoice());

        ((CustomView) findViewById(R.id.customView)).setTouchActionListener(new CustomView.TouchActionListener() {
            @Override
            public void onTouchDown(int x, int y) {
                switch (notifcationType) {
                    case TOAST:
                        Toast.makeText(MainActivity.this, "Нажаты координаты [" + x + " : " + y + "]", Toast.LENGTH_SHORT).show();
                        break;
                    case SNACKBAR:
                        Snackbar.make(findViewById(R.id.parentContainer), "Нажаты координаты [" + x + " : " + y + "]", BaseTransientBottomBar.LENGTH_SHORT).show();
                        break;
                    }
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.activity_main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int itemID = item.getItemId();
        if(itemID == R.id.chooseNotificationType){
            startNotificationTypeActivity();
        }
        return super.onOptionsItemSelected(item);
    }

    private void startNotificationTypeActivity(){
        Intent intent = new Intent(MainActivity.this, NotificationTypeActivity.class);
        startActivity(intent);
    }
}

File: FoodApp/app/src/main/java/com/example/foodapp/viewprimarypage/MainActivity.kt
package com.example.foodapp.viewprimarypage

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import com.example.foodapp.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), OnItemCategoryClickListener, OnItemAreaClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbarPrimaryPage)

        showFragmentCategory()
        showFragmentArea()
        showFragmentRandomDishPrimary()
        floatingButton.setOnClickListener { showFragmentRandomDishDisplayDish() }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when(item.itemId){
            R.id.navigateToFavoriteDishesActivity -> startFavoriteDishesActivity()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun startFavoriteDishesActivity(){
        val instance = FavoriteDishesActivity.newInstance()
        val intent = instance.newIntent(this)
        startActivity(intent)
    }

    private fun showFragmentCategory() {
        supportFragmentManager.beginTransaction()
            .add(
                R.id.fragmentContainerCategory,
                FragmentCategory.newInstance(),
                FragmentCategory.TAG
            )
            .commit()
    }

    private fun showFragmentArea() {
        supportFragmentManager.beginTransaction()
            .add(
                R.id.fragmentContainerArea,
                FragmentArea.newInstance(),
                FragmentArea.TAG
            )
            .commit()
    }

    private fun showFragmentRandomDishPrimary() {
        supportFragmentManager.beginTransaction()
            .add(
                R.id.fragmentContainerRandomDish,
                FragmentRandomDishPrimary.newInstance(),
                FragmentRandomDishPrimary.TAG
            )
            .commit()
    }

    private fun showFragmentRandomDishDisplayDish() {
        supportFragmentManager.beginTransaction()
            .add(
                R.id.fragmentContainerRandomDish,
                FragmentRandomDishDisplayDish.newInstance(),
                FragmentRandomDishDisplayDish.TAG
            )
            .commit()
    }

    override fun displayItemCategory(categoryName: String) {
        val categoryListActivity = CategoryListActivity.newInstance()
        val intent = categoryListActivity.newIntent(this)
        intent.putExtra(KEY_EXTRA_CATEGORY_NAME, categoryName)
        startActivity(intent)
    }

    override fun displayItemArea(areaName: String) {
        val areaListActivity = AreaListActivity.newInstance()
        val intent = areaListActivity.newIntent(this)
        intent.putExtra(KEY_EXTRA_AREA_NAME, areaName)
        startActivity(intent)
    }
}